### DISCUSSION TITLE
How autoinstall works

### DISCUSSION THREAD

**Discussion by Weasleman** [vmware/packer-examples-for-vsphere#623] [Help]
I am looking to use packer to run ubuntu 20.04. Now I understand how packer works high level with preseed.cfg file but trying to understand the auto install. 

I get that you have two files user-data and meta-data. But not sure if i am gripping this fully.

Like the preseed you mount the meta-data and user-data using either a floppy or CD rom within packer.
example:

floppy_files = [
   "./preseed.cfg"
]

Now looking at the vsphere-iso_basic_ubuntu.pkr.hcl we have boot commands and various other commands. So im guessing that packer does these steps. Connect to vsphere uses the ISO installs then runs through the user-data file. this sets up the operating system.

I am working backwards using this as a starting point i think this maybe why im getting confused

https://askubuntu.com/questions/1385563/packer-ubuntu-server-20-04-prompting-for-install-confirmation-despite-autoinsta

Just want some help understanding.

**Comment by tenthirtyam** [comment]
Moved to the Discussions as this is not a documentation issue specifically for this project and is more general Packer / Packer Plugin questions.

**Reply by Weasleman** [reply]
Apologies

**Reply by tenthirtyam** [reply]
This question would be better for [discuss.hashicorp.com](https://discuss.hashicorp.com/c/packer/23.) or the [Discord](https://discord.com/channels/884074247289438210/884076234320011264) community.
